client.go

```diff
@@ -595,3 +655,54 @@ func (client *client) updateMetadata(data *MetadataResponse) ([]string, error) {
 	}
 	return ret, err
 }
+
+func (client *client) getConsumerMetadata(consumerGroup string, attemptsRemaining int) (*ConsumerMetadataResponse, error) {
+	for broker := client.any(); broker != nil; broker = client.any() {
+
+		Logger.Printf("client/coordinator Requesting coordinator for consumergoup %s from %s.\n", consumerGroup, broker.Addr())
+
+		request := new(ConsumerMetadataRequest)
+		request.ConsumerGroup = consumerGroup
+
+		response, err := broker.GetConsumerMetadata(request)
+
+		if err != nil {
+			Logger.Printf("client/coordinator Request to broker %s failed: %s.\n", broker.Addr(), err)
+
+			switch err.(type) {
+			case PacketEncodingError:
+				return nil, err
+			default:
+				_ = broker.Close()
+				client.deregisterBroker(broker)
+				continue
+			}
+		}
+
+		switch response.Err {
+		case ErrNoError:
+			Logger.Printf("client/coordinator Coordinator for consumergoup %s is #%d (%s:%d).\n", consumerGroup, response.CoordinatorID, response.CoordinatorHost, response.CoordinatorPort)
+			return response, nil
+
+		case ErrConsumerCoordinatorNotAvailable:
+			Logger.Printf("client/coordinator Coordinator for consumer group %s not yet available, trying again in %dms...\n", consumerGroup, client.conf.Metadata.Retry.Backoff/time.Millisecond)
+			time.Sleep(2 * time.Second)
+			continue
```

Comment:
I thought we'd had this conversation already :P

There are two retriable error cases here:
- offsets topic has not yet been created
- offsets topic exists but an election is ongoing so there is no leader right now

The first case returns `ErrConsumerCoordinatorNotAvailable`. It takes some time (~5s?) to create the topic. If the topic fails to create for some reason, waiting another 5 seconds is unlikely to fix anything.

I don't know what the second case returns (`ErrConsumerCoordinatorNotAvailable` or `LeaderNotAvailable` or ...?). This one is worth retrying several times in fairly quick succession (100-500ms) since leadership time depends on how busy the cluster is but is usually fairly fast.